sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -206,86 +209,48 @@ private Request CreateAuthRequest(MsiType msiType, string[] scopes, string clien
             };
         }
 
-        private bool ImdsAvailable(CancellationToken cancellationToken)
+        public virtual bool ImdsAvailable(CancellationToken cancellationToken)
         {
-            // send a request without the Metadata header.  This will result in a failed request,
-            // but we're just interested in if we get a response before the timeout of 500ms
-            // if we don't get a response we assume the imds endpoint is not available
-            using (Request request = _pipeline.HttpPipeline.CreateRequest())
+            // try to create a TCP connection to the IMDS IP address. If the connection can be established
+            // we assume that IMDS is available. If connecting times out or fails to connect assume that
+            // IMDS is not available in this environment.
+            try
             {
-                request.Method = RequestMethod.Get;
-
-                request.Uri.Reset(s_imdsEndpoint);
-
-                request.Uri.AppendQuery("api-version", ImdsApiVersion);
-
-                CancellationToken imdsTimeout = new CancellationTokenSource(ImdsAvailableTimeoutMs).Token;
-
-                try
+                using (var client = new TcpClient())
                 {
-                    Response response = _pipeline.HttpPipeline.SendRequest(request, CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, imdsTimeout).Token);
+                    var result = client.BeginConnect(s_imdsHostIp, s_imdsPort, null, null);
```

Comment:
I think it would be better to use Connect.

Reply:
After looking into this we can't use `Connect` and `ConnectAsync` because they don't provide a timeout or CancellationToken parameter.